JahsieAult/homework.md

```diff
@@ -0,0 +1,164 @@
+
+**m)** `++x == x++`
+>True. If X is a number value the addition takes place after the comparison. If the value of x is not a number this will be false.
+
+**n)** `"1"+x == 1+x`
+>False. If X is a number it will concatenate the first pair and add the second. If X is a string it will contcatenate both sides and be true.
+
+**o)** `"0"+1 == 1`
+>True. "0" is interpreted as a number, and 1+0=1
+
+**p)** `(typeof (x+1))===(typeof x)`	
+>If X is a number, string or NaN it evaluates to true. False if X is a boolean, undefined, or null. The console registers (x+1) as a number. 
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`
+>True unless X is NaN. If X is NaN both sides are false, but for all other value types at least one side of the statement will be true.
+**r)** `(x=(typeof (x+(typeof x))))==x`
+>True. You are assigning X to be something and then checking to see if X is equal to it. The answer will always be true. 
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+ **2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.
+>(x + y + z) / 3
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+
+>var mean = (x + y + z) / 3;
+x = x + (mean/2 - x);
+y = y + (mean/2 - y);
+z = z + (mean/2 -z);
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+>(Not sure that I understood the placement of the variables) 
+var
+var area = (t - b) * (r - l)
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+>(t - b) > (r - l)
+
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+>if ((t - b) > (r - l)); {
+	var maxCircumference = 3.14 * ((r - l)/2); 
+	} else {
+		var maxCircumference = 3.14 * ((t - b)/2);
+	}
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
```

Comment:
3e) In order to subdivide the rectangle as the problem asks, we need values equal to 1/3 the length and width. We can then add those values to the existing sides and get the sides of the new one.